plugin/evm/limit_order.go

```diff
@@ -128,30 +129,45 @@ func (lop *limitOrderProcesser) runMatchingEngine(longOrders []limitorders.Limit
 }
 
 func (lop *limitOrderProcesser) runLiquidations(market limitorders.Market, longOrders []limitorders.LimitOrder, shortOrders []limitorders.LimitOrder) (filteredLongOrder []limitorders.LimitOrder, filteredShortOrder []limitorders.LimitOrder) {
-	markPrice := lop.memoryDb.GetLastPrice(market)
-	var oraclePrice float64 = 20 // @todo: change this
-
-	liquidableTraders := lop.memoryDb.GetLiquidableTraders(market, markPrice, oraclePrice)
-
-	for _, liquidable := range liquidableTraders {
-		var matchedOrders []limitorders.LimitOrder
-		if liquidable.Size > 0 {
-			// we'll need to sell, so we need a long order to match
-			matchedOrders = longOrders
-		} else {
-			matchedOrders = shortOrders
+	oraclePrice := big.NewInt(20 * 10e6) // @todo: get it from the oracle
+
+	longPositions, shortPositions := lop.memoryDb.GetLiquidableTraders(market, oraclePrice)
+
+	for i, liquidable := range longPositions {
+		if len(shortOrders) == 0 {
+			log.Error("no matching order found for liquidation", "trader", liquidable.Address.String(), "size", liquidable.Size)
+			continue // so that all other liquidable positions get logged
 		}
-		if len(matchedOrders) == 0 {
+		for j, shortOrder := range shortOrders {
+			if liquidable.GetUnfilledSize().Sign() == 0 {
+				break
+			}
+			fillAmount := utils.BigIntMinAbs(liquidable.GetUnfilledSize(), shortOrder.GetUnFilledBaseAssetQuantity())
+			if fillAmount.Sign() == 0 {
+				continue
+			}
+			lop.limitOrderTxProcessor.ExecuteLiquidation(liquidable.Address, shortOrder, fillAmount)
+			shortOrders[j].FilledBaseAssetQuantity.Sub(shortOrders[j].FilledBaseAssetQuantity, fillAmount)
+			longPositions[i].FilledSize.Add(longPositions[i].FilledSize, fillAmount)
```

Comment:
can be simplified to
```
shortOrders[j].FilledBaseAssetQuantity.Sub(fillAmount)
longPositions[i].FilledSize.Add(fillAmount)
```

Reply:
https://cs.opensource.google/go/go/+/go1.19.5:src/math/big/int.go;l=115

these functions accept exactly 2 args

Reply:
I thought so too, but changed it to that and didn't run into compilation error

Reply:
It didn't let me do it 
<img width="773" alt="Screenshot 2023-01-19 at 3 15 00 PM" src="https://user-images.githubusercontent.com/22051142/213409263-118cc2da-e2fa-450c-824c-271246499b28.png">